### ISSUE TITLE
Provide way to turn on badges for Fenix installation testing

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/addons-frontend#9912]
Relates to #9878

Fenix are looking to test installation in AMO ahead of enabling it. 

Whilst this is tested, to avoid turning it on on AMO and having Fenix clients getting a link they can't use before the feature is enabled on Fenix, this bug is to add a feature to allow turning on the button conditionally on, in production for testing Fenix.  

This could potentially be enabled via a URL param.

/cc @jvillalobos

**Comment by willdurand** [comment]
@muffinresearch would it be possible for Fenix to use -dev?

**Comment by muffinresearch** [comment]
> @muffinresearch would it be possible for Fenix to use -dev?

I'll let @csadilek confirm, but I think that there would still be a need to test in prod. If Fenix is like desktop, it would be limited to testing with dev/stage certs only on nightly - but it's a good point - maybe that's not an issue?

**Comment by csadilek** [comment]
@muffinresearch @willdurand We're limiting add-ons installation (from content) to requests coming from `addons.mozilla.org` to prevent installation on any other domains.

We could including staging and dev here, but I was told it would be tricky for you to make sure the add-on and file IDs line up in staging/dev. The ID in the URL needs to map to the add-on/file ID of the recommended list/collection used in Fenix e.g. in `https://addons.mozilla.org/android/downloads/file/3683403/search_by_image-3.4.1-an+fx.xpi` -> `3683403` would also have to the ID for the file in the collection used by Fenix to make it installable.

So far we've tested by enabling the button manually in the debugger and confirmed it will work :). I don't think we need a permanent or complex solution here as we can enable this in a few weeks anyway. So, I'd say, if this is tricky for you let's just do another manual test instead.

**Comment by willdurand** [comment]
> We could including staging and dev here, but I was told it would be tricky for you to make sure the add-on and file IDs line up in staging/dev.

Yes, you'd have to have "Fenix collections" created in -dev/-stage too but that does not sound like a bad idea given the tight integration between Fenix and AMO.

Talking about collections, if a user uses [a different collection in Nightly](https://blog.mozilla.org/addons/2020/09/29/expanded-extension-support-in-firefox-for-android-nightly/), is the "recommended" list still loaded?

**Comment by csadilek** [comment]
> Talking about collections, if a user uses a different collection in Nightly, is the "recommended" list still loaded?

Currently it's the configured collection that defines what add-ons are supported/recommended in Fenix. There's a separate discussion around this and how to continue going forward. This case here is only about installing the officially recommended add-ons from AMO so I think we can keep the Nightly override case separate, as it's only available in the developer menu and for advanced users i.e. it allows users to install any add-on they like in Nightly.

**Comment by willdurand** [comment]
> Currently it's the configured collection that defines what add-ons are supported/recommended in Fenix. There's a separate discussion around this and how to continue going forward. This case here is only about installing the officially recommended add-ons from AMO so I think we can keep the Nightly override case separate, as it's only available in the developer menu and for advanced users i.e. it allows users to install any add-on they like in Nightly.

So in this case, maybe the "install from AMO button link" shouldn't be enabled in Fenix when the collection is not the default/recommended collection? Otherwise AMO will enable buttons that won't work as expected in Fenix.

**Comment by csadilek** [comment]
> Yes, you'd have to have "Fenix collections" created in -dev/-stage too but that does not sound like a bad idea given the tight integration between Fenix and AMO.

It depends what we're trying to accomplish with this ticket. If it's just about a one-time test, the yes, I could create a custom build (using AMO staging) and ask QA to test it or we (our teams) can use it ourselves, but then we might as well test manually one more time.

I think the goal here was to test this on the official Nightly|Beta builds of Fenix which use production servers to fetch the add-on collection. So, it would be nice if we could enable the button with a request parameter or have any other test page to use.

**Comment by jvillalobos** [comment]
I don't think we should worry about users with a custom collection. AMO behavior should only reflect what the Recommended list dictates.

**Comment by willdurand** [comment]
> I don't think we should worry about users with a custom collection. AMO behavior should only reflect what the Recommended list dictates.

I agree. Fenix should support add-ons in the recommended list _in addition to_ the add-ons in the custom collection.

**Comment by bobsilverberg** [comment]
I feel like this issue is now talking about multiple things, but perhaps I am wrong about that. 

The first issue is the one that this was originally opened about, which is to provide a way to temporarily enable the install buttons on production for testing purposes. We could do that via a URL param, but it sounds like the manual way of testing that (by enabling the button via the debugger) is also feasible. @csadilek how would you like us to proceed? It's not an insignificant amount of work to add the URL param trigger, so if your team is fine with continuing to test manually that would be our preference.

It sounds like a second issue was introduced which discusses whether we should be enabling the install buttons not only for add-ons that are officially supported on Fenix, but also for add-ons in a custom collection. I believe our initial take on that was that we needn't support those in a custom collection, but after the discussion above I feel that needs clarification.

@csadilek and @jvillalobos what are your thoughts on that?

**Comment by csadilek** [comment]
> I feel like this issue is now talking about multiple things, but perhaps I am wrong about that.

@bobsilverberg yes, two different topics here:

> how would you like us to proceed? It's not an insignificant amount of work to add the URL param trigger, so if your team is fine with continuing to test manually that would be our preference.

If this is a lot of work for your team, plus something that's a burden to maintain going forward, then in my opinion, it's fine to take the risk and rely on manual testing. We've already done two rounds and this should work once enabled. We should also land an integration test for it in Fenix once enabled to guard against regressions. If something changes on your end between now and January (e.g. the Url to the .xpi file), please let us know, as we will need changes in Fenix.

> I believe our initial take on that was that we needn't support those in a custom collection, but after the discussion above I feel that needs clarification.

We are aware of the edge case for users with custom collection and discussed it with @jvillalobos. In the short-term, we agreed that this limitation is acceptable, especially as users with custom collections can install any add-on they like in Fenix directly and will also get a meaningful error message when trying to install an add-on from AMO that's not in the their collection. We already started working through the details of switching to the new API for recommended extensions which should ultimately replace collections in Fenix.

**Comment by willdurand** [comment]
@bobsilverberg yes, you're right: 2 different topics and it was too late last Friday for me to clarify about these two topics.

We won't deploy AMO until next year (2021.01.07 to be more precise) so I guess manual testing is fine too. In addition, we have -dev and -stage environments to try things like this usually. We _could_ enable the feature (install buttons enabled for Fenix) on -stage if needed.

> I believe our initial take on that was that we needn't support those in a custom collection, but after the discussion above I feel that needs clarification.

@bobsilverberg This is definitely not an AMO issue.

> In the short-term, we agreed that this limitation is acceptable, especially as users with custom collections can install any add-on they like in Fenix directly and will also get a meaningful error message when trying to install an add-on from AMO that's not in the their collection. We already started working through the details of switching to the new API for recommended extensions which should ultimately replace collections in Fenix.

@csadilek thanks. My understanding is that Fenix Nightly will tell the user that an install from AMO isn't possible because they configured a custom collection (after they clicked an install button on AMO). Is that correct?

**Comment by csadilek** [comment]
@willdurand yes, these users will get an error message displayed after they clicked the button. Note that this affects very few users and only on Nightly, plus they already can install any add-on they like. But even so, this problem should be temporary until both our teams find a way to integrate the new API.

**Comment by muffinresearch** [comment]
Closing this in favour of the manual testing route proposed in https://github.com/mozilla/addons-frontend/issues/9912#issuecomment-744638322
